main.tf

```
@@ -105,7 +146,14 @@ resource "azuread_application" "this" {
 }
 
 resource "random_uuid" "oauth2_permission_scope" {
-  count = length(var.oauth2_permission_scopes)
+  // Generate random UUIDs for each permission scope
+  for_each = var.oauth2_permission_scopes
+}
+
+resource "random_uuid" "app_role" {
+  // Generate random UUIDs for each app role
+  // We use app role value as key for since it is the most likely value to be unique
+  for_each = { for role in var.app_roles : role.value => role }
```

Am I wrong to think that it would be a more robust solution to change the type of variable `app_roles` to `map(...)` rather than `list(...)`, similar to variable `oauth2_permission_scopes`, and then make the following change?

```suggestion
  for_each = var.app_roles
```

---

I agree, will add the change.